server-adapters/resteasy-reactor-netty/src/main/java/org/jboss/resteasy/plugins/server/reactor/netty/ReactorNettyJaxrsServer.java

```diff
@@ -0,0 +1,209 @@
+
+   final Logger log = LoggerFactory.getLogger(ReactorNettyJaxrsServer.class);
+
+   protected String hostname = null;
+   protected int configuredPort = PortProvider.getPort();
+   protected int runtimePort = -1;
+   protected ResteasyDeployment deployment;
+   protected String root = "";
+   protected SecurityDomain domain;
+   private SSLContext sslContext;
+   private EmbeddedServerHelper serverHelper = new EmbeddedServerHelper();
+
+   private DisposableServer server;
+
+   @Override
+   public ReactorNettyJaxrsServer deploy() {
+      return this;
+   }
+
+   @Override
+   public ReactorNettyJaxrsServer start() {
+      log.info("Starting RestEasy Reactor-based server!");
+      serverHelper.checkDeployment(deployment);
+
+      final HttpServer svrBuilder = HttpServer.create().port(configuredPort);
+      if (hostname != null && !hostname.trim().isEmpty()) {
+         svrBuilder.host(hostname);
+      }
+      final Handler handler = new Handler();
+      server = svrBuilder
+          .handle(handler::handle)
+          .bindNow();
+      runtimePort = server.port();
+
+      server.onDispose().block();
+      return this;
+   }
+
+   class Handler {
+
+      private final Mono<InputStream> empty = Mono.just(new InputStream() {
+         @Override
+         public int read() {
+            return -1;  // end of stream
+         }
+      });
+
+      Publisher<Void> handle(final HttpServerRequest req, final HttpServerResponse resp) {
+         final ResteasyUriInfo info = new ResteasyUriInfo(req.uri(), "/");
+
+         // aggregate (and maybe? asInputStream) reads the entire request body into memory (direct?)
+         // Can we stream it in some way?
+         // https://stackoverflow.com/a/51801335/2071683 but requires a thread.  Isn't using a thread
+         // per request even if from the elastic pool a big problem???  I mean we are trying to reduce
+         // threads!
+         // I honestly don't know what netty is doing here.  When I try to send a large body it says
+         // "request payload too large".  I don't know if that's configurable or not..
+        
+
+         // This is a subscription tied to the completion writing the response.
+         final MonoProcessor<Void> completionMono = MonoProcessor.create();
+
+         return req.receive()
+             .aggregate()
+             .asInputStream()
+             .switchIfEmpty(empty)
+             .flatMap(body -> {
+                 // These next 2 classes provide the main '1-way bridges' between reactor-netty and RestEasy.
+                 final ReactorNettyHttpResponse resteasyResp =
+                     new ReactorNettyHttpResponse(req.method().name(), resp, completionMono);
+
+                 final ReactorNettyHttpRequest resteasyReq =
+                     new ReactorNettyHttpRequest(info, req, body, resteasyResp, (SynchronousDispatcher) deployment.getDispatcher());
```

Comment:
I need to check what `SynchronousDispatcher` vs async.  I guess sync one just executes within the same thread instead of jumping to a different thread pool?